### PR TITLE
Add suport for link meta

### PR DESCRIPTION
**Pull request by pandermatt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,7 +232,7 @@ You can take advantage of the publish mechanism of Dato CMS to expire the cache.
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. 
+After checking out the repo, run `bundle install` to install dependencies.
```
Comment (pandermatt) [inline README.md:235]: i couldn't find bin/setup in the project 😢

File: README.md [hunk 2]
```diff
@@ -245,7 +245,7 @@ to consume data from your project.
 
 Then, run `bundle exec rspec` to run the tests. 
 
-You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+You can also run `rails console` for an interactive prompt that will allow you to experiment.
```
Comment (pandermatt) [inline README.md:248]: i couldn't find bin/console in the project 😢

File: bin/rails [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
-APP_PATH = File.expand_path("../test/dummy/config/application", __dir__)
+ENGINE_PATH = File.expand_path("../lib/dato/engine", __dir__)
+APP_PATH = File.expand_path("../spec/dummy/config/application", __dir__)
 
 # Set up gems listed in the Gemfile.
 ENV["BUNDLE_GEMFILE"] ||= File.expand_path("../Gemfile", __dir__)
 require "bundler/setup" if File.exist?(ENV["BUNDLE_GEMFILE"])
 
-require "rails/all"
+require "rails"
```
Comment (pandermatt) [inline bin/rails:13]: i needed to change this; otherwise it always asked me for a DB when calling `rails console`

Comment (pandermatt) [reply bin/rails:13]: yet `rails s` does not run, with this changes i get at least another error

File: app/components/dato/link.rb [hunk 4]
```diff
@@ -8,4 +8,23 @@ def initialize(node, root)
   def generated_tag
     "a"
   end
+
+  def link_attributes
+    attr = {
+      "href" => @node.url,
+      "class" => "dato-cms-#{@node.type}",
+    }
+    %w[rel target].each do |type|
```
Comment (coorasse) [inline app/components/dato/link.rb]: are these the only possible meta?

File: bin/rails (hunk 3 above)
Comment (coorasse) [reply bin/rails:13]: I don't think you are supposed to run the dummy rails app. This is meant for testing

File: spec/components/dato/link_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    is_expected.to have_selector("a.dato-cms-link span.dato-cms-span", text: "The best company!")
+  end
+
+  context "when link has no meta" do
+    let(:meta) { [] }
+
+    it "does not render meta" do
+      is_expected.to have_selector("a.dato-cms-link[href='https://renuo.ch/']")
+      is_expected.not_to have_selector("a.dato-cms-link[rel='nofollow']")
+      is_expected.not_to have_selector("a.dato-cms-link[target='_blank']")
```
Comment (coorasse) [inline spec/components/dato/link_spec.rb]: can we rather check that rel and target attributes are not set?

**Review by coorasse — CHANGES_REQUESTED** [review]
Thanks for your contribution! I have some suggestions/questions

File: bin/rails (hunk 3 above)
Comment (pandermatt) [reply bin/rails:13]: Ok!

File: app/components/dato/link.rb (hunk 4 above)
Comment (pandermatt) [reply app/components/dato/link.rb]: I only found those in the document

https://www.datocms.com/docs/structured-text/dast#link

File: app/components/dato/link.rb [hunk 6]
```diff
@@ -8,4 +8,23 @@ def initialize(node, root)
...
+      attr[type] = value if value
+    end
+    attr
+  end
+
+  private
+
+  def extract_meta(type)
+    meta = @node.meta.find { |m| m.id == type }
+    meta&.value
```
Comment (coorasse) [inline app/components/dato/link.rb]:
```suggestion
    @node.meta.find { |m| m.id == type }&.value
```

File: app/components/dato/link.rb [hunk 7]
```diff
@@ -8,4 +8,23 @@ def initialize(node, root)
...
   end
+
+  def link_attributes
+    attr = {
+      "href" => @node.url,
+      "class" => "dato-cms-#{@node.type}",
+    }
+    %w[rel target].each do |type|
+      value = extract_meta(type)
+      attr[type] = value if value
```
Comment (coorasse) [inline app/components/dato/link.rb]: is it necessary to check `if value`? could we simply always set it? maybe compact empty values at the end?

File: spec/dummy/config/database.yml
Comment (coorasse) [inline spec/dummy/config/database.yml:1]: is this file committed by mistake?

**Review by coorasse — COMMENTED** [review]
Thanks Pascal. I still have some improvements proposals

File: app/components/dato/link.rb (hunk 7 above)
Comment (pandermatt) [reply app/components/dato/link.rb]: yes true! when it's nil, its sent as nil :D

**Review by coorasse — APPROVED** [review]
Thanks!

**Pull request merged**